packages/ragbits-document-search/tests/integration/test_unstructured.py

```diff
@@ -33,7 +33,7 @@ async def test_document_processor_processes_text_document_with_unstructured_prov
 
     elements = await document_processor.get_provider(document_meta).process(document_meta)
 
-    assert isinstance(document_processor._providers[DocumentType.TXT], UnstructuredDefaultProvider)
+    assert isinstance(document_processor._providers[DocumentType.TXT], type(UnstructuredDefaultProvider))
```

Comment:
Adding some tests that would make sure that both kinds of configurations (type-based and instance-based) work correctly with `DocumentProcessorRouter` would be nice

Reply:
done